integration-tests/src/test/java/oracle/weblogic/kubernetes/ItIstioSessionMigrIstioEnabled.java

```diff
@@ -0,0 +1,402 @@
+   *                   JUnit engine parameter resolution mechanism
+   */
+  @BeforeAll
+  public static void initAll(@Namespaces(2) List<String> namespaces) {
+    logger = getLogger();
+
+    // get a new unique opNamespace
+    logger.info("Assign unique namespace for Operator");
+    assertNotNull(namespaces.get(0), "Namespace list is null");
+    opNamespace = namespaces.get(0);
+
+    logger.info("Assign unique namespace for Domain");
+    assertNotNull(namespaces.get(1), "Namespace list is null");
+    domainNamespace = namespaces.get(1);
+
+    // Label the domain/operator namespace with istio-injection=enabled
+    Map<String, String> labelMap = new HashMap();
+    labelMap.put("istio-injection", "enabled");
+    assertDoesNotThrow(() -> addLabelsToNamespace(domainNamespace,labelMap));
+    assertDoesNotThrow(() -> addLabelsToNamespace(opNamespace,labelMap));
+
+    // install and verify operator
+    installAndVerifyOperator(opNamespace, domainNamespace);
+
+    List<String> appList = new ArrayList();
+    appList.add(SESSMIGR_APP_NAME);
+
+    // build the model file list
+    final List<String> modelList = Collections.singletonList(MODEL_DIR + "/" + SESSMIGR_MODEL_FILE);
+
+    // create image with model files
+    logger.info("Create image with model file and verify");
+    String miiImage = createMiiImageAndVerify(SESSMIGR_IMAGE_NAME, modelList, appList);
+
+    // docker login and push image to docker registry if necessary
+    dockerLoginAndPushImageToRegistry(miiImage);
+
+    // config the domain with Istio ingress with Istio gateway
+    String managedServerPrefix = domainUid + "-managed-server";
+    assertDoesNotThrow(() ->
+        configIstioModelInImageDomain(miiImage, domainNamespace, domainUid, managedServerPrefix),
+        "setup for istio based domain failed");
+
+    // map to save HTTP response data
+    httpAttrMap = new HashMap<String, String>();
+    httpAttrMap.put("sessioncreatetime", "(.*)sessioncreatetime>(.*)</sessioncreatetime(.*)");
+    httpAttrMap.put("sessionid", "(.*)sessionid>(.*)</sessionid(.*)");
+    httpAttrMap.put("primary", "(.*)primary>(.*)</primary(.*)");
+    httpAttrMap.put("secondary", "(.*)secondary>(.*)</secondary(.*)");
+    httpAttrMap.put("count", "(.*)countattribute>(.*)</countattribute(.*)");
+  }
+
+  /**
+   * When Istio is enabled, test sends a HTTP request to set http session state(count number),
+   * get the primary and secondary server name, session create time and session state and from the util method
+   * and save HTTP session info, then stop the primary server by changing ServerStartPolicy to NEVER and
+   * patching domain. Send another HTTP request to get http session state (count number), primary server
+   * and session create time. Verify that a new primary server is selected and HTTP session state is migrated.
+   */
+  @Test
+  @DisplayName("When Istio is enabled, stop the primary server, verify that a new primary server is picked and "
```

Comment:
Display name can be shortened  --> Verify session migration in an istio enabled environment

Reply:
changed